.travis.yml

```diff
@@ -0,0 +1,126 @@
+#=============================================================================
+# Travis CoreNeuron settings
+#=============================================================================
+
+#=============================================================================
+# Environment
+#=============================================================================
+language: cpp
+
+#=============================================================================
+# Build matrix
+#=============================================================================
+matrix:
+  fast_finish: true
+  include:
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=ON"
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=OFF"
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=ON"
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=OFF"
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_NMODL=ON"
+        - PYTHON_VERSION=3.6.7  # for nmodl pip3 install
+        - USE_NMODL=ON
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=ON"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=OFF"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=ON"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=OFF"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_NMODL=ON"
+        - USE_NMODL=ON
+    - os: linux
+      dist : xenial
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - bison
+            - g++-9
+            - flex
+            - libopenmpi-dev
+            - openmpi-bin
+      env:
+        - GCC_VERSION=9
+#=============================================================================
+# Common Packages
+#=============================================================================
+addons:
+  # for Linux builds, we use APT
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      - bison
+      - flex
+      - libopenmpi-dev
+      - openmpi-bin
+      - libboost-all-dev
```

Comment:
@WeinaJi : do we need to duplicate `addons` section here? I see that some of those being defined at line 61.